app/designer/features/props-panel/use-props-logic.test.ts

```diff
@@ -0,0 +1,265 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+import { components, UserProp } from "@webstudio-is/sdk";
+import { SelectedInstanceData } from "~/shared/canvas-components";
+import { usePropsLogic } from "./use-props-logic";
+
+const getSelectedInstanceData = (
+  componentName: keyof typeof components,
+  props: UserProp[]
+): SelectedInstanceData => {
+  return {
+    id: "2",
+    component: componentName,
+    cssRules: [],
+    browserStyle: {},
+    props: {
+      id: "1",
+      props,
+      instanceId: "2",
+      treeId: "1",
+    },
+  };
+};
+
+describe("usePropsLogic", () => {
+  test("should return required props", () => {
+    const { result } = renderHook(() =>
+      usePropsLogic({
+        selectedInstanceData: getSelectedInstanceData("Link", []),
+        publish: jest.fn(),
+      })
+    );
+    expect(result.current.userProps.length).toEqual(1);
+    expect(result.current.userProps[0]).toMatchObject({
+      prop: "href",
+      value: "",
+      required: true,
+    });
+  });
+
+  test.only("should return different default props for different instances", () => {
```

Comment:
forgotten .only, need to add a lint rule for this

Reply:
removed